server/schedule/hbstream/heartbeat_stresms_test.go

```diff
@@ -0,0 +1,66 @@
+package hbstream
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/kvproto/pkg/eraftpb"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/tikv/pd/pkg/mock/mockcluster"
+	"github.com/tikv/pd/pkg/mock/mockhbstream"
+	"github.com/tikv/pd/server/config"
+)
+
+func TestHeaertbeatStreams(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testHeartbeatStreamSuite{})
+
+type testHeartbeatStreamSuite struct {
+}
+
+func (s *testHeartbeatStreamSuite) TestActivity(c *C) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	cluster := mockcluster.NewCluster(config.NewTestOptions())
+	cluster.AddRegionStore(1, 1)
+	cluster.AddRegionStore(2, 0)
+	cluster.AddLeaderRegion(1, 1)
+	region := cluster.GetRegion(1)
+	msg := &pdpb.RegionHeartbeatResponse{
+		ChangePeer: &pdpb.ChangePeer{Peer: &metapb.Peer{Id: 2, StoreId: 2}, ChangeType: eraftpb.ConfChangeType_AddLearnerNode},
+	}
+
+	hbs := NewHeartbeatStreams(ctx, 1, cluster)
+	stream1, stream2 := mockhbstream.NewHeartbeatStream(), mockhbstream.NewHeartbeatStream()
+
+	// Active stream is stream1.
+	hbs.BindStream(1, stream1)
+	time.Sleep(50 * time.Millisecond)
```

Comment:
Is there any better way to notify ? `Sleep` maybe not so stable.

Reply:
Seems not easy to do because heartbeat stream provides an async interface. Maybe I should use `testutil.WaitUtil` to make it more predictable.